app/models/adviser/sign_up.rb

```diff
@@ -22,11 +27,21 @@ def save
 
     AdviserSignUpWorker.perform_async(application_form.id, preferred_teaching_subject_id)
 
+    application_form.update!(signed_up_for_adviser: true)
+
     true
   end
 
   def available?
-    feature_active? && application_form.valid?
+    return false unless feature_active?
+
+    refresh_adviser_status_from_git_api
+
+    application_form_validations.valid?
+  rescue GetIntoTeachingApiClient::ApiError => e
+    Rails.logger.warn("Adviser sign up unavailable due to API error: #{e.message}")
+
+    false
```

Comment:
I think we could move this whole method and it's other method dependencies outside of this model (doing the request, cache in Redis and rescuing the API error). Maybe call `Adviser::SignUpAvailability` that returns true or false?